packages/orchestrator-ui-components/src/components/WfoRouteChangeListener/WfoRouteChangeListener.tsx

```diff
@@ -0,0 +1,22 @@
+import React, { useEffect } from 'react';
+
+import { getSession, signOut } from 'next-auth/react';
+import { useRouter } from 'next/router';
+
+import { useGetOrchestratorConfig } from '@/hooks';
+
+export const WfoRouteChangeListener = () => {
+    const router = useRouter();
+    const { authActive } = useGetOrchestratorConfig();
+
+    useEffect(() => {
+        if (authActive) {
+            getSession().then((session) => {
```

Comment:
Try to stick with the async await pattern, we have that in most places in this lib

Reply:
We have .thens in other places